Selector template

taysea opened the pull request:
<!--- Provide a general summary of the PR in the Title above -->


<!--- Insert the PR's # for the deploy preview's URL -->
[Deploy Preview](https://deploy-preview-3865--keen-mayer-a86c8b.netlify.app/templates/selector)

#### What does this PR do?

- [x] What should keyboard behavior be? Tab or arrows.
   - Tabs, this component behaves most like a checkboxgroup, so align to native behavior.
- [x] Selector `title` , `description`, `icon` config -- here or on SelectorHeader?
   - On Selector is fine for template, people can expand in their repo if they find a need or we can expand in future.
- [x] If you click on quick filter after having filtered/searched by something, should it clear those other filters/search out?
   - Yes, quick filters essentially act as an exposed saved view.
- [x] Do we need to supportive cases where there's interactive content inside?
   - Not at this time. Let's let use cases drive the conversation before we build to account for it.

2 outstanding decisions we need to make. 

- [x] Wrapping within the Selector. Currently `wrap` is used to allow the titlte/description to wrap under the icon when space is limited. That said, sometimes some Selectors wrap before others in the same group just based on how long their title is. Are we okay with this? Is that misalignment awkward? Should the wrapping be breakpoint driven (that feels a little awkward)? Other approaches?
    - After looking into this, `wrap` isn't the best solution because it wasn't allowing the title to wrap at all. Removing `wrap`. I'm wondering at this point if we need to support the icon on top of title variant at this point?
- [x] Right now the selectors are content driven, so in the grid, one row might grow taller than another just based on the content. Is this okay? If we wanted all rows consistent we'd need to set a fixed row height, but that feels dangerous/might cause content to clip.

Screen image to reflect the above issues

<img width="1073" alt="Screen Shot 2024-05-21 at 1 42 01 PM" src="https://github.com/grommet/hpe-design-system/assets/42451602/e59b4e94-6b5e-423c-a4cd-872e0f3d80d5">


#### Where should the reviewer start?

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### How should this be manually tested?

#### Any background context you want to provide?

#### What are the relevant issues?
Closes https://github.com/grommet/hpe-design-system/issues/3840, https://github.com/grommet/hpe-design-system/issues/3867

#### Screenshots (if appropriate)

#### Should this PR be mentioned in Design System updates?

#### Is this change backwards compatible or is it a breaking change?

#### Notifications

[New]Selector

[]

[]

jcfilben commented on aries-site/src/examples/components/selector/SupportSelector.js:
```diff
@@ -0,0 +1,135 @@
+    value: 'aruba',
+    title: 'Aruba AP-635 WiFi 6E',
+    description: <Text size="xlarge">18</Text>,
+    icon: <Iteration height="medium" />,
```
```suggestion
    icon: <Iteration aria-hidden height="medium" />,
```
Since the icons are decorative lets go ahead and mark them with `aria-hidden` to give a better screenreader experience.

jcfilben commented on aries-site/src/examples/components/selector/SupportSelector.js:
It feels like this example is right on the line of the maximum number of items we would want to display in this type of format. Like if we have any more than 8 maybe there is a better, more space conscious, component that should be used instead. I think if we publish this page it may be helpful to have a little more guidance on when to use.
<img width="1077" alt="Screen Shot 2024-05-20 at 3 21 08 PM" src="https://github.com/grommet/hpe-design-system/assets/54560994/65d38d21-afb5-4703-bf57-1a1b356a3e7f">

Looks like we need to make some adjustments on the smaller screen size. I'm wondering if it makes sense to use a select/selectmultiple when we switch down to mobile view and there are more than ~3 options.
<img width="449" alt="Screen Shot 2024-05-20 at 3 22 49 PM" src="https://github.com/grommet/hpe-design-system/assets/54560994/e23a881c-663b-4172-93c5-40445def4d78">

britt6612 replied:
Definitely should have some sort of guidance for sure based on the number of selectors. 

>> It feels like this example is right on the line of the maximum number of items we would want to display in this type of format. Like if we have any more than 8 maybe there is a better, more space conscious, component that should be used instead. I think if we publish this page it may be helpful to have a little more guidance on when to use.

For now we just moved this to be single column for small and xsmall break points. We can discuss if we want to enhance in future to do a select/selectmultiple

jcfilben replied:
The small and extra small breakpoints look good now, but it looks like there is still some responsive issues.

<img width="761" alt="Screen Shot 2024-05-22 at 4 14 12 PM" src="https://github.com/grommet/hpe-design-system/assets/54560994/fe863a18-b873-406d-9329-fbfa2cd32f98">

jcfilben commented on aries-site/src/examples/components/selector/SupportSelector.js:
The text wrapping is sometimes a little awkward. I wonder how it would look if the second card's text started on the line right next to the icon (like the first card) but then just took up 2 lines?

Not a huge deal but I'm confused what the '18' is supposed to mean in these examples.
<img width="543" alt="Screen Shot 2024-05-20 at 3 27 17 PM" src="https://github.com/grommet/hpe-design-system/assets/54560994/f34e9ab7-9163-4980-a4d2-191d2158e5f7">

taysea replied:
Good callout, that's one of our remaining questions about how to handle wrapping in this instance.

Re: "18", it's supposed to represent how many "Aruba AP-..." there are but I can see it's confusing since the data repeats itself. @britt6612 can you update to be unique/more realistic?

britt6612 replied:
yes I can!

britt6612 replied:
updated

jcfilben commented on aries-site/src/examples/components/selector/QuickFilter.js:
This isn't necessarily an issue but I just want to flag it to make sure we are aware of it before we move forward. The green outline is a little similar to our focus indicator. I don't think this is an accessibility issue, but just want to make sure from a styling perspective we are okay with this.

<img width="808" alt="Screen Shot 2024-05-20 at 3 30 27 PM" src="https://github.com/grommet/hpe-design-system/assets/54560994/b828cf2a-bc44-41c8-8b30-23caff807382">

<img width="809" alt="Screen Shot 2024-05-20 at 3 32 55 PM" src="https://github.com/grommet/hpe-design-system/assets/54560994/128daf77-dd2f-4f66-be3a-94cd77d168ad">

britt6612 replied:
This is also how our Secondary Button is do I think its fine but valid call out

jcfilben commented on aries-core/src/js/components/core/Selector/SelectorGroup.js:
```diff
@@ -0,0 +1,131 @@
+    <ThemeContext.Extend
+      // theme overrides to remove excess padding
+      value={{
+        checkBox: { pad: 'none' },
```
It looks like CheckBox has a pad prop, does it work to just set that to 'none'?

jcfilben commented on aries-site/src/examples/components/selector/QuickFilter.js:
When I use voice over and I read all the elements on the page (ctrl + option + a) as opposed to tabbing through I am getting a weird behavior where the failed launches radio button is read out as '1 of 19' and successful launches is '2 of 19'. It seems like there is an issue in the way the screen reader is interpreting the groups. I would expect it to announce as '1 of 2' or '2 or 2'

britt6612 replied:
So I heard this as well its actually counting all of the RadioButtons on the whole page not just in that group. Im not sure if this is expected or why its counting all of them.

taysea replied:
Can someone add a screen recording? I can't replicate this.

taysea replied:
This is resolved with `aria-hidden="true"` since the radiobutton control is purely decorative.

jcfilben commented on aries-site/src/data/structures/components.js:
```diff
@@ -1392,6 +1393,46 @@ export const components = [
+      actions: [
+        {
+          label: 'See source code',
+          icon: <ShareRounded />,
```
```suggestion
          icon: <ShareRounded aria-hidden />,
```

britt6612 commented:
2 outstanding decisions we need to make. 

- [ ] Wrapping within the Selector. Should we add this so that there is columns at certain points vs row or leave as is for now. 
- [ ] Right now the selectors are content driven is this okay? Should we have them all look consistent across? 

Screen image to reflect the above issues

<img width="1073" alt="Screen Shot 2024-05-21 at 1 42 01 PM" src="https://github.com/grommet/hpe-design-system/assets/42451602/e59b4e94-6b5e-423c-a4cd-872e0f3d80d5">

taysea commented:
We're getting WAVE violations since the radiobutton/checkbox doesn't have a label. In our case, we really just want the radiobutton / checkbox as a visual but it's not actually performing any form logic. Thoughts on how we might resolve? @jcfilben @britt6612 
 
<img width="1663" alt="Screen Shot 2024-05-22 at 2 44 58 PM" src="https://github.com/grommet/hpe-design-system/assets/12522275/e5b3cba4-0b20-4e8f-9b64-453b90d3b4ea">

jcfilben commented:
> We're getting WAVE violations since the radiobutton/checkbox doesn't have a label. In our case, we really just want the radiobutton / checkbox as a visual but it's not actually performing any form logic. Thoughts on how we might resolve? Do we need to worry about this since it's not actually interactive? Could we associate it with the title? @jcfilben @britt6612

Instead of using the CheckBox/RadioButton component can we just have an icon?

jcfilben commented on aries-site/src/examples/components/selector/SupportSelector.js:
```diff
@@ -0,0 +1,101 @@
+            <Heading margin="none">What can we help you with?</Heading>
+            <Text>Choose a product to continue.</Text>
+          </Box>
+          <TextInput
+            aria-label="search"
+            icon={<SearchIcon />}
+            placeholder="Search all assets"
+            type="search"
+            width="medium"
+          />
```
Can we remove the search here since it is not functional?

taysea replied:
Is it valuable to have if someone were to copy + paste this template?

taysea commented:
> > We're getting WAVE violations since the radiobutton/checkbox doesn't have a label. In our case, we really just want the radiobutton / checkbox as a visual but it's not actually performing any form logic. Thoughts on how we might resolve? Do we need to worry about this since it's not actually interactive? Could we associate it with the title? @jcfilben @britt6612
> 
> When I run DevTools it is flagging the RadioButton/CheckBox as nested interactive elements. So that is another issue in addition to the label problem.
> 

Hmm .. even with the tabIndex={-1}? I thought that was a viable pattern for this kind of thing.

> I feel like this case isn't necessarily an accessibility issue, but it also doesn't align with the WCAG rules so kinda a grey area.
> 
> Instead of using the CheckBox/RadioButton component can we just have an icon? This might just create a cleaner experience in general. Even with `aria-hidden` set on the RadioButton/CheckBox we still do technically have two nested click targets. We also have this focus behavior coming through on RadioButton if you click directly on it.
> 

With regards to use of an icon, that would have to be a design discussion we bring up with design team. Lucas had two design versions of this component (one with just an icon, one with the RadioButton/Checkbox) and had settled on the RadioButton/CheckBox I believe as a result of some user testing, so we should better understand if the icon path might still be an option.


> <img alt="Screen Shot 2024-05-22 at 4 20 35 PM" width="392" src="https://private-user-images.githubusercontent.com/54560994/332977483-2ba34af9-edc5-4685-b02a-33089dea7c7c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0MTY4NDYsIm5iZiI6MTcxNjQxNjU0NiwicGF0aCI6Ii81NDU2MDk5NC8zMzI5Nzc0ODMtMmJhMzRhZjktZWRjNS00Njg1LWIwMmEtMzMwODlkZWE3YzdjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIyVDIyMjIyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkyOWI1NzBjNzAwNDBmMGJmNjFmN2ZhZTg5ODJlMDMyMTk3ODZkOTE1YzZjNTFkNGI4YjAwZDY0N2MyY2VhNzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KMA6oZXyfijPivPL8Lr19vEvAMFZoCAJeUnC_Pfs4-Y">

Also wasn't expected this to be able to be active element with tabIndex, but realizing that doesn't necessarily block "click". That said, we do a similar pattern on DS site for our preview cards which is why I thought it'd work.

britt6612 commented:
> > > We're getting WAVE violations since the radiobutton/checkbox doesn't have a label. In our case, we really just want the radiobutton / checkbox as a visual but it's not actually performing any form logic. Thoughts on how we might resolve? Do we need to worry about this since it's not actually interactive? Could we associate it with the title? @jcfilben @britt6612
> > 
> > 
> > When I run DevTools it is flagging the RadioButton/CheckBox as nested interactive elements. So that is another issue in addition to the label problem.
> 
> Hmm .. even with the tabIndex={-1}? I thought that was a viable pattern for this kind of thing.
> 
> > I feel like this case isn't necessarily an accessibility issue, but it also doesn't align with the WCAG rules so kinda a grey area.
> > Instead of using the CheckBox/RadioButton component can we just have an icon? This might just create a cleaner experience in general. Even with `aria-hidden` set on the RadioButton/CheckBox we still do technically have two nested click targets. We also have this focus behavior coming through on RadioButton if you click directly on it.
> 
> With regards to use of an icon, that would have to be a design discussion we bring up with design team. Lucas had two design versions of this component (one with just an icon, one with the RadioButton/Checkbox) and had settled on the RadioButton/CheckBox I believe as a result of some user testing, so we should better understand if the icon path might still be an option.
> 
> > <img alt="Screen Shot 2024-05-22 at 4 20 35 PM" width="392" src="https://private-user-images.githubusercontent.com/54560994/332977483-2ba34af9-edc5-4685-b02a-33089dea7c7c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0MTY4NDYsIm5iZiI6MTcxNjQxNjU0NiwicGF0aCI6Ii81NDU2MDk5NC8zMzI5Nzc0ODMtMmJhMzRhZjktZWRjNS00Njg1LWIwMmEtMzMwODlkZWE3YzdjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIyVDIyMjIyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkyOWI1NzBjNzAwNDBmMGJmNjFmN2ZhZTg5ODJlMDMyMTk3ODZkOTE1YzZjNTFkNGI4YjAwZDY0N2MyY2VhNzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KMA6oZXyfijPivPL8Lr19vEvAMFZoCAJeUnC_Pfs4-Y">
> 
> Also wasn't expected this to be able to be active element with tabIndex, but realizing that doesn't necessarily block "click". That said, we do a similar pattern on DS site for our preview cards which is why I thought it'd work.

I almost wonder if we can just make an icon look like our radiobutton and checkbox then use that if that is what is wanted in the design.

jcfilben reviewed (commented):
Can we round the Checkbox icon container so that it matches the hpe theme styling for the CheckBox component?

britt6612 commented:
> Can we round the Checkbox icon container so that it matches the hpe theme styling for the CheckBox component?

done

jcfilben commented on aries-core/src/js/components/core/Selector/SelectorGroup.js:
```diff
@@ -0,0 +1,142 @@
+  );
+
+  return (
+    <ThemeContext.Extend
+      // theme overrides to remove excess padding
+      value={{
+        checkBox: { pad: 'none' },
+        radioButton: {
+          hover: {
+            background: 'none',
+          },
+          container: {
+            extend: '',
+          },
+          extend: '',
+        },
+      }}
+    >
```
Can we remove this theme override since we are using icons now?

britt6612 replied:
good call out yes!

jcfilben commented:
> Wrapping within the Selector. Currently wrap is used to allow the titlte/description to wrap under the icon when space is limited. That said, sometimes some Selectors wrap before others in the same group just based on how long their title is. Are we okay with this? Is that misalignment awkward? Should the wrapping be breakpoint driven (that feels a little awkward)? Other approaches?
> After looking into this, wrap isn't the best solution because it wasn't allowing the title to wrap at all. Removing wrap. I'm wondering at this point if we need to support the icon on top of title variant at this point?
> 

I don't feel like we need to support the icon on top variant

> Right now the selectors are content driven, so in the grid, one row might grow taller than another just based on the content. Is this okay? If we wanted all rows consistent we'd need to set a fixed row height, but that feels dangerous/might cause content to clip.

I think it is fine if one row is taller than another row

britt6612 commented:
Just an update on the changes that Matt added:

SelectorGroup: 
```
  children,
  defaultValue,
  layout = 'grid',
  multiple,
  onSelect,
  value,
```

The new change here is the `layout` this is a new added prop and it takes either `grid` `fit`. When the `grid` is used then the `columns` are set to `count: 'fit', size: ['small', 'flex']` while the `fit` is wrapped in a Box with `direction='row'`

Selector:

The changes here are the following condition was added 
```
      {(indicator !== false || title || description || icon) && (
          <SelectorHeader
            indicator={indicator}
            title={title}
            icon={icon}
            description={description}
            selected={selected}
          />
        )}
```

SelectorHeader:

Here the new addition was you can pass an object to SelectorIndicator for example 
`indicator={{ round: 'full', size: 'medium' }}`
so when `multiple` if someone wanted their indicator to be round that can happen. 

Matt left a comment if there should always be a `checkmark` which him Jessica and I did discuss however we need to run by Lucas in my opinion to make sure this is okay on the design side as well.

jcfilben commented:
> Here the new addition was you can pass an object to SelectorIndicator for example
indicator={{ round: 'full', size: 'medium' }}
so when multiple if someone wanted their indicator to be round that can happen.

I'm confused why we added this. I was thinking having the square vs circle was included so that people could have a visual indicator of whether the selector is a single select or multiple. Is there a use case we are envisioning for passing an object to `indicator` to override this behavior?

jcfilben commented on aries-core/src/js/components/core/Selector/SelectorHeader.js:
```diff
@@ -0,0 +1,82 @@
+      {...rest}
+    >
+      {selected && (
+        // need to talk with Lucas
```
I'm not opposed to this, I like how it looks visually, but it seems like we lose a little of the styling similarities between radio button and single selection. This change might not make it is as clear for the user that the circle is a single select.

I agree it would be good to check with Lucas

britt6612 replied:
yes Im planning tomorrow to discuss

britt6612 commented on aries-site/src/pages/components/selector.mdx:
```diff
@@ -0,0 +1,78 @@
+  TEMP,
+} from '../../examples';
+
+{/* Let's move this under templates instead of components */}
```
oh shoot need to do this thanks

britt6612 replied:
actually @taysea didnt we say we wanted this under components? Thats why we were adding the notification?

taysea replied:
Matt decided it would be better to have it in Templates for now. Let's go ahead with moving it there for now.

britt6612 replied:
okay was there any reasoning behind that or?

taysea replied:
His reasoning was to keep "components" limited to things that truly have a published Grommet component. If we lean into that, we should move "Search" to templates as well (currently under components)

britt6612 replied:
☝️ thank you that was my next question was should we move "Search"

britt6612 replied:
moved to templates

britt6612 commented:
> > Here the new addition was you can pass an object to SelectorIndicator for example
> > indicator={{ round: 'full', size: 'medium' }}
> > so when multiple if someone wanted their indicator to be round that can happen.
> 
> I'm confused why we added this. I was thinking having the square vs circle was included so that people could have a visual indicator of whether the selector is a single select or multiple. Is there a use case we are envisioning for passing an object to `indicator` to override this behavior?

good question, I also don't have clarity when reviewing I figure it was just Matts way of making things more flexible. I wonder if he saw another use case with clicking multiple but then had a round indicator.

taysea commented:
> > > Here the new addition was you can pass an object to SelectorIndicator for example
> > > indicator={{ round: 'full', size: 'medium' }}
> > > so when multiple if someone wanted their indicator to be round that can happen.
> > 
> > 
> > I'm confused why we added this. I was thinking having the square vs circle was included so that people could have a visual indicator of whether the selector is a single select or multiple. Is there a use case we are envisioning for passing an object to `indicator` to override this behavior?
> 
> good question, I also don't have clarity when reviewing I figure it was just Matts way of making things more flexible. I wonder if he saw another use case with clicking multiple but then had a round indicator.

@jcfilben @britt6612 I believe this was to allow something a bit more flexible 

> > > Here the new addition was you can pass an object to SelectorIndicator for example
> > > indicator={{ round: 'full', size: 'medium' }}
> > > so when multiple if someone wanted their indicator to be round that can happen.
> > 
> > 
> > I'm confused why we added this. I was thinking having the square vs circle was included so that people could have a visual indicator of whether the selector is a single select or multiple. Is there a use case we are envisioning for passing an object to `indicator` to override this behavior?
> 
> good question, I also don't have clarity when reviewing I figure it was just Matts way of making things more flexible. I wonder if he saw another use case with clicking multiple but then had a round indicator.

@jcfilben @britt6612 I believe this was to give a bit more flexibility to support something like Gina's "choice chips" design if desired.

jcfilben commented on aries-site/src/examples/cardPreviews/selector.js:
Can we add a little gap between the selector elements on the Card preview? They just look a little squished
<img width="494" alt="Screen Shot 2024-06-18 at 12 21 48 PM" src="https://github.com/grommet/hpe-design-system/assets/54560994/f95bde31-92f5-428c-ba60-46f04768bae5">

britt6612 replied:
good catch

britt6612 replied:
fixed in last commit

jcfilben reviewed (commented):
Just one more comment, otherwise looks good

jcfilben reviewed (approved):
Looks good!

julauxen commented on aries-site/src/data/structures/templates/templates.js:
```diff
@@ -222,6 +223,29 @@ export const templates = [
+    cardOrder: 4,
+    description: `A selection control that allows users to choose one or
+    more related options to access information, filter, make selections
+    or take actions.`,
```
I would not say "take actions" because we had a lot of confusion in terms of that functionality. The selector is not a button, therefore not a place to take action.

britt6612 replied:
@vavalos5 lets work on re-wording

julauxen commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,93 @@
+
+### Quick Filtering
+
+Enables users to filter data based on high-level visual options to narrow down datasets within tables and list. Making it quickly recognizable, based on specific criteria, enhancing efficiency and usability.
```
"Enables users to filter data based on high-level visual options to narrow down datasets within tables and list."

All the rest is not really adding clarity to the use case.

britt6612 replied:
```suggestion
Enables users to filter data based on high-level visual options to narrow down datasets within tables and lists.
```

julauxen commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,93 @@
+
+### Quick option selection
+
+Helps users quickly navigate through choices, ensuring they find the relevant options with ease.
```
"Enables users to quickly navigate through choices, ensuring they find the relevant options with ease."

Just changed a bit to keep the format of text of previous use case.

julauxen commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,93 @@
+</Example>
+
+## Anatomy
+
```
There's no imagine for anatomy? It's hard to understand without a visual

vavalos5 replied:
Totally get it. We have an image in Figma and something that's still a bit more TBD. Imagery will come later.

britt6612 replied:
anatomy to come..

britt6612 replied:
have to code it up

julauxen commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,93 @@
+
+| Label | Region                  | Purpose                                                      | Required | Notes                                                   |
+| :---: | --------------------    | ------------------------------------------------------------ | :------: | ------------------------------------------------------- |
+| **1** | **Title**               | Heading providing context to the options available.          | Optional | Titles should use a Heading of the correct semantic heading level based on where the selector is in the document. |
```
Is the title optional?

I ask because it's weird to me reading the guidance that everything in a selector is optional.

vavalos5 replied:
@britt6612 I'll let you input here

britt6612 replied:
So essentially the way Selector is built in code is that a person does not necessarily have to provide a title..

<img width="1087" alt="Screen Shot 2024-07-02 at 11 31 14 AM" src="https://github.com/grommet/hpe-design-system/assets/42451602/90c76fdc-fc20-44c9-8cbe-3984dde6cf25">


for example above the user is passing a text within `children` which has a different placement for the title. We built selector in code to be very flexible so we are not making anything `required` because the user can really pass a variety of things into the Selector

julauxen commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,93 @@
+| **5** | **Selector Indicator**  | Visual cue (such as a checkmark or highlight) to denote selected items. | Optional | aria-label required. |
+
+## Types
+
```
I would suggest we use a different visual representation for "Types", like a table with a visual and an explanation, so the user can easily understand the difference not only on functionality but on type of component.

Also, if we are delivering all the examples on the TEMP kitchen sink, we might need to amplify this section, since we have different visual types for single select and multi select.

vavalos5 replied:
I'm not sure I'm following. For types we don't include imagery but rather indicate the "types" of selectors there are.

For kitchen sink, these are all TBD and will be clarified soon. :)

britt6612 replied:
Kitchen sink is just for us devs to test will not be published

britt6612 replied:
@julauxen Im also not sure I follow your visual with "types"

britt6612 commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,93 @@
+
+### Quick option selection
+
+Helps users quickly navigate through choices, ensuring they find the relevant options with ease.
```
```suggestion
Enables users to quickly navigate through choices, ensuring they find the relevant options with ease.
```

taysea commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,93 @@
+
+## Use cases
+
+### Quick Filtering
```
```suggestion
### Quick filtering
```
Sentence case

taysea commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,93 @@
+
+### Quick Filtering
+
+Enables users to filter data based on high-level visual options to narrow down datasets within tables and lists.
```
It feels like this should capture:
- Provides an efficient way to filter data to a specific view that will be useful and relevant to the workflow.

Maybe not the best wording, but given its high visual prominence, they should be useful and relevant.

It feels like the current guidance is redefining "Selector" in some ways (high prominence visual options) / we can clean up wording since "to narrow down datasets" is redundant with "filter data".

britt6612 replied:
okay I re-wording lmk what you think

taysea commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,93 @@
+
+### Quick option selection
+
+Enables users to quickly navigate through choices, ensuring they find the relevant options with ease.
```
Since this is "use cases" which should be a bit specific. Can we add something that says this is typically found in support and onboarding flows?

britt6612 replied:
done

taysea commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,93 @@
+
+### Single select
+
+Users can select only one option at a time. Ideal for scenarios where only one choice is allowed.
```
Let's reuse similar wording from ToggleGroup:

With single selection, only one option can be selected at a time. This is the default behavior.

britt6612 replied:
done

taysea commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,93 @@
+
+### Multiple select
+
+Users can select multiple options simultaneously. Useful for filtering or multi-choice settings.
```
```suggestion
With multiple selection, multiple options can be selected at the same time.
```

vavalos5 commented on aries-site/src/pages/templates/selector.mdx:
```diff
@@ -0,0 +1,98 @@
+| Label  | Region                  | Purpose                                                                                      | Required | Notes                                                                                                                                                                    |
+| :----: | ----------------------- | -------------------------------------------------------------------------------------------- | :------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| **1**  | **Container**           | Provides distinct boundary representing the option as a distinct choice.                     | Required | --                                                                                                                                                                       |
+| **2**  | **Selection Indicator** | Provides additional visual prominence and distinction between selected and unselected state. | Optional | For single it will be rounded for multiple sqaured off                                                                                                                   |
```
for the notes, spell check: multiple squared off
